### ISSUE TITLE
SDL_SetWindowFullscreenMode causes segmentation fault

### ISSUE THREAD

**Issue by Synt4xErr0r4** [libsdl-org/SDL#10431]
Hello,

I was just trying to use `SDL_SetWindowFullscreenMode`, but it just kept segfaulting, so I looked into it a bit.
GDB says that the memory passed to `SDL_memcmp` is not accessible.

So I looked at the source for `SDL_GetFullscreenModeMatch`, which called `SDL_memcmp`.
I'm pretty sure the conditions of the two if statements in [`SDL_GetFullscreenModeMatch`](https://github.com/libsdl-org/SDL/blob/main/src/video/SDL_video.c#L1144) are wrong:

```c
        /* Search for an exact match */
        if (!mode) {
            for (int i = 0; display->num_fullscreen_modes; ++i) { //< here...
                if (SDL_memcmp(&fullscreen_mode, &display->fullscreen_modes[i], sizeof(fullscreen_mode)) == 0) {
                    mode = &display->fullscreen_modes[i];
                    break;
                }
            }
        }

        /* Search for a mode with the same characteristics */
        if (!mode) {
            for (int i = 0; display->num_fullscreen_modes; ++i) { //< ...and here
                if (cmpmodes(&fullscreen_mode, &display->fullscreen_modes[i]) == 0) {
                    mode = &display->fullscreen_modes[i];
                    break;
                }
            }
        }
```

They should probably be `i < display->num_fullscreen_modes` instead if I'm not mistaken.

**Comment by slouken** [comment]
Yep, good catch, thanks!

**Comment by slouken** [comment]
Fixed in 89de8e211
